File: Mobile/owner/lib/models/branch.dart
class Branch {
  final int id;
  final String name;
  final String code;
  final String? address;
  final String? phone;
  final String? email;
  final String? gstNumber;
  final bool isActive;

  Branch({
    required this.id,
    required this.name,
    required this.code,
    this.address,
    this.phone,
    this.email,
    this.gstNumber,
    this.isActive = true,
  });

  factory Branch.fromJson(Map<String, dynamic> json) {
    return Branch(
      id: json['id'],
      name: json['name'],
      code: json['code'],
      address: json['address'],
      phone: json['phone'],
      email: json['email'],
      gstNumber: json['gst_number'],
      isActive: json['is_active'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'code': code,
      'address': address,
      'phone': phone,
      'email': email,
      'gst_number': gstNumber,
      'is_active': isActive,
    };
  }
}

File: ResortApp/app/models/branch.py
from sqlalchemy import Column, Integer, String, Text, Boolean, DateTime, text
from datetime import datetime
from app.database import Base

class Branch(Base):
    __tablename__ = "branches"
    
    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)          # e.g., "Orchid Main", "Orchid Beach"
    code = Column(String, unique=True, nullable=False, index=True) # e.g., "MAIN", "BEACH"
    address = Column(Text, nullable=True)
    phone = Column(String, nullable=True)
    email = Column(String, nullable=True)
    gst_number = Column(String, nullable=True)
    image_url = Column(String, nullable=True)      # Main image for the branch/resort
    facebook = Column(String, nullable=True)
    instagram = Column(String, nullable=True)
    twitter = Column(String, nullable=True)
    linkedin = Column(String, nullable=True)
    is_active = Column(Boolean, nullable=False, default=True, server_default=text('true'))
    created_at = Column(DateTime, nullable=False, default=datetime.utcnow, server_default=text('CURRENT_TIMESTAMP'))
    
    def __repr__(self):
        return f"<Branch id={self.id} name={self.name} code={self.code}>"
